### PR TITLE
Update readme following move to PaaS

### PR DESCRIPTION
**Pull request by stevehook**
## Context

We moved our infrastructure from Azure to PaaS. There were still a few references to Azure in our README and other docs. This PR attempts to update those. It only adds minimal info about PaaS - I think there's scope for a lot more detail in some of the linked docs but I'm going to leave that for another PR (and maybe somebody more knowledgeable).

There should be no code changes in this PR, it's documentation only.

## Changes proposed in this pull request

- [x] Update the README itself
- [x] Update the architecture diagram on the README
- [x] Update the linked Azure Infrastructure doc and diagrams (and rename it PaaS Infrastructure)

Still to do in separate PR(s):

- Include information about permissions in the PaaS Infrastructure doc (what is the equivalent of PIM)?
- Update the developer onboarding docs (include links to the relevant PaaS/CloudFoundary docs).
- Search for any other references to Azure in the docs.

## Guidance to review

- What information about PaaS would be helpful to other devs?
- Anything missing?

## Link to Trello card

https://trello.com/c/HBRjdUM0/3364-update-readme-following-paas-migration

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,13 +8,13 @@ A service for candidates to [apply for teacher training](https://www.apply-for-t
...
+| ---------- | -------------------------------------------------------------------- | ----------------------------------------------------------------------- | ------------- |----------------- |
+| Production | [www](https://www.apply-for-teacher-training.service.gov.uk)         | Public site                                                             | `bat-prod`    | `apply-prod`     |
+| Staging    | [staging](https://staging.apply-for-teacher-training.service.gov.uk) | For internal use by DfE to test deploys                                 | `bat-staging` | `apply-staging`  |
+| Sandbox    | [sandbox](https://sandbox.apply-for-teacher-training.service.gov.uk) | Demo environment for software vendors who integrate with our API        | `bat-prod`    | `apply-sandbox`  |
+| QA         | [qa](https://qa.apply-for-teacher-training.service.gov.uk)           | For internal use by DfE for testing. Automatically deployed from master | `bat-qa`      | `apply-qa`       |
+| DevOps     | [dev](https://dev.apply-for-teacher-training.service.gov.uk)         | For testing infrastructure changes                                      |               |                  |
```
Comment (duncanjbrown) [inline README.md:17]: Does this still exist?

Comment (stevehook) [reply README.md:17]: Was wondering the same thing. Doesn't look like it, I'm going to remove it for now...

File: docs/developer-onboarding.md [hunk 2]
```diff
@@ -2,22 +2,12 @@
...
 
-This document describes the on-boardng steps that need to be undertaken for new Developers when they join the team.
+This document describes the on-boardng steps for new Developers when they join the team.
 
-## Azure/DevOps steps
+## PaaS/DevOps steps
 
-These steps can only be completed by members of the DevOps community. At the present time the contacts for this are Colin Saliceti, Vignesh Natraj or Karthik Kumar.
+### Get a PaaS account
+You can get an account by requesting one in the `#digital-tools-support` Slack channel. Request an account with a SpaceDeveloper role for your @digital.education.gov.uk email address with access to the `dfe` organisation and `bat-prod`, `bat-staging` and `bat-qa` spaces.
```
Comment (stevehook) [inline docs/developer-onboarding.md:11]: Should this page have a quick tutorial on PaaS? Or should it have links to a good resource for learning PaaS?

File: docs/paas-infrastructure.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## PaaS spaces
+
+The apply service is deployed across three PaaS 'spaces' that are shared with other services in BAT.
+
+- `apply-prod` contains the *production* and *sandbox* Apply environment.
+- `apply-staging` contains the *staging* Apply environment.
+- `apply-qa` contains the *qa* Apply environment.
+
+## Resource permissions
```
Comment (stevehook) [inline docs/paas-infrastructure.md:15]: For Azure this section described PIM. Is there anything we need to say about PaaS and permissions here?

File: docs/paas-infrastructure.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+## Application resource group
+
+The diagram below shows the core application dependent resources contained within each of the environments listed above. The resources allocated to each application will vary across the environments depending upon the individual demands on their infrastructure, but the actual components are identical.
+
+![PaaS Infrastructure](paas-infrastructure.svg)
```
Comment (stevehook) [inline docs/paas-infrastructure.md:21]: This diagram is just a few boxes with the obvious moving parts of the system in. Is there other more useful information we can include.

File: docs/developer-onboarding.md (hunk 2 above)
Comment (malcolmbaig) [reply docs/developer-onboarding.md:11]: A link to https://docs.cloud.service.gov.uk/ would be helpful, but I think a cheatsheet with some common commands and a mention of `make <env_name> shell` would be a nice addition as well.

File: README.md [hunk 5]
```diff
@@ -8,13 +8,12 @@ A service for candidates to [apply for teacher training](https://www.apply-for-t
...
-| DevOps     | [dev](https://dev.apply-for-teacher-training.service.gov.uk)         | For testing infrastructure changes                                      | `s106d02` |
+| Name       | URL                                                                  | Description                                                             | PaaS space    | PaaS application |
+| ---------- | -------------------------------------------------------------------- | ----------------------------------------------------------------------- | ------------- |----------------- |
+| Production | [www](https://www.apply-for-teacher-training.service.gov.uk)         | Public site                                                             | `bat-prod`    | `apply-prod`     |
+| Staging    | [staging](https://staging.apply-for-teacher-training.service.gov.uk) | For internal use by DfE to test deploys                                 | `bat-staging` | `apply-staging`  |
+| Sandbox    | [sandbox](https://sandbox.apply-for-teacher-training.service.gov.uk) | Demo environment for software vendors who integrate with our API        | `bat-prod`    | `apply-sandbox`  |
```
Comment (cwrw) [inline README.md:15]: maybe we can move sandbox to be after prod just because they are in the same space?

File: docs/paas-infrastructure.md (hunk 3 above)
Comment (cwrw) [reply docs/paas-infrastructure.md:15]: We're planning to discuss if we want to follow something similar to PIMs vs give everyone access as it stands, hopefully in the coming week, I think for now we can leave this out. Another section I can think of are secrets in Azure, do we want to add a section around those?

File: docs/tech-architecture.svg [hunk 6]
```diff
@@ -1 +1 @@
...-1.015625 0.375 -1.78125q0.390625 -0.765625 1.078125 -1.1875q0.703125 -0.421875 1.640625 -0.421875q0.609375 0 1.140625 0.1875q0.546875 0.1875 0.953125 0.59375q0.421875 0.40625 0.65625 1.03125q0.234375 0.625 0.234375 1.484375l0 0.484375l-5.359375 0l0 -1.046875l3.890625 0q0 -0.4375 -0.1875 -0.78125q-0.1875 -0.359375 -0.53125 -0.546875q-0.328125 -0.203125 -0.78125 -0.203125q-0.46875 0 -0.828125 0.234375q-0.359375 0.21875 -0.5625 0.59375q-0.1875 0.359375 -0.1875 0.796875l0 0.90625q0 0.578125 0.203125 0.984375q0.203125 0.40625 0.578125 0.625q0.375 0.203125 0.890625 0.203125q0.328125 0 0.59375 -0.09375q0.28125 -0.09375 0.484375 -0.28125q0.203125 -0.1875 0.296875 -0.46875l1.4375 0.171875q-0.125 0.5625 -0.515625 0.984375q-0.375 0.421875 -0.96875 0.65625q-0.59375 0.234375 -1.359375 0.234375zm4.221405 -0.125l0 -6.546875l1.484375 0l0 1.09375l0.078125 0q0.171875 -0.5625 0.609375 -0.875q0.4375 -0.3125 1.0 -0.3125q0.125 0 0.28125 0.015625q0.171875 0 0.28125 0.03125l0 1.421875q-0.109375 -0.046875 -0.328125 -0.0625q-0.203125 -0.03125 -0.390625 -0.03125q-0.421875 0 -0.765625 0.1875q-0.328125 0.171875 -0.53125 0.5q-0.1875 0.3125 -0.1875 0.734375l0 3.84375l-1.53125 0z" fill-rule="nonzero"/></g></svg>
```
Comment (cwrw) [inline docs/tech-architecture.svg:1]: what changed here as I can still see azure in the final image?

**Review by cwrw — COMMENTED** [review]
Thanks for updating this! Some minor suggestions inline

File: docs/paas-infrastructure.md (hunk 4 above)
Comment (cwrw) [reply docs/paas-infrastructure.md:21]: maybe something around diagnostics and logging we send to [grafana](https://grafana-bat.london.cloudapps.digital/d/eF19g4RZx/cf-apps)/logit and alerting, although maybe those are better suited in their own section

File: docs/tech-architecture.svg (hunk 6 above)
Comment (stevehook) [reply docs/tech-architecture.svg:1]: that's odd, I didn't chance much but it should look like this...
image

Comment (cwrw) [reply docs/tech-architecture.svg:1]: bah I think Github might be displaying the old one when I try viewing it in this pr 👍

File: docs/paas-cheatsheet.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+If you want to see back a little further back into the logs:
+```
+$ cf logs <APP_NAME> --recent
+```
+
+## Shelling into a container
+To shell into a container:
+```
+$ cf ssh <APP_NAME>
```
Comment (despo) [inline docs/paas-cheatsheet.md:72]:
```suggestion
$ cf ssh <APP_NAME>-clock
```



SSHing into a clock or worker should avoid hanging sessions bumping up the memory (as mentioned on this thread: https://ukgovernmentdfe.slack.com/archives/CN1MCQCHZ/p1620639185388500)

**Review by cwrw — APPROVED** [review]

**Pull request merged**